doc/release/upcoming_changes/17029.compatibility.rst

```diff
@@ -0,0 +1,23 @@
+Casting errors interrupt Iteration
+----------------------------------
+NumPy sometimes has issues reporting errors during casting
+correctly. This is rare, since the vast majority of casts
+will never result in errors.
+However, in some cases of casting errors less of the result
+may be written back. The state of arrays involved in errors
+should *always* be considered undefined.
+Possible changes occur for:
+
+* Universal functions when casting is unsafe (rare)
+* Advanced index assignments
+* The lowlevel `numpy.nditer` API.
+
```

Comment:
```suggestion
When errors occur during casting the casting procedure will now
be fully stopped in most cases.  This means that in some cases
less of the resulting data will be processed.  This always returned
undefined and partial results.
```
@mattip I will suggest at least this. The C-API note probably doesn't affect many or anyone, but I feel something should be said...

Reply:
How about "When iterating while casting values, an error may stop the iteration earlier than before. In any case, a failed casting operation always returned undefined, partial results. Those may now be even more undefined and partial."

Reply:
Sounds good.